Documentation/gettingstarted/encryption-wireguard.rst

```diff
@@ -0,0 +1,111 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _encryption_wg:
+
+********************************
+Wireguard Transparent Encryption
+********************************
+
+This guide explains how to configure Cilium with transparent encryption of
+traffic between Cilium-managed endpoints using `Wireguard <https://www.wireguard.com/>`_.
+
+When Wireguard is enabled in Cilium, the agent running on each cluster node
+will establish a secure Wireguard tunnel between it and all other known nodes
+in the cluster. Each node automatically creates its own encryption key-pair and
+distributes its public key via the ``io.cilium.network.wg-pub-key`` annotation in
+the Kubernetes ``CiliumNode`` custom resource object. Each node's public key is then
+used by other nodes to decrypt and encrypt traffic from and to Cilium-managed
+endpoints running on that node.
```

Comment:
I think in the upcoming architecture section we should extend this saying that the priv key is stored in /var/run/cilium and it does not survive node restarts.